x-pack/plugins/alerting/server/alert/alert.ts

```diff
@@ -276,10 +276,6 @@ export class Alert<
       return false;
     }
 
-    return !summarizedAlerts.all.data.some(
-      (alert) =>
-        get(alert, ALERT_INSTANCE_ID) === this.getId() ||
-        get(alert, ALERT_RULE_UUID) === this.getId()
-    );
+    return !summarizedAlerts.all.data.some((alert) => get(alert, ALERT_UUID) === this.getUuid());
```

Comment:
@ersin-erdal are we all good now?

Reply:
Unfortunately this doesn't work for Sec Rules. (Works for observability rule though)
UUID of the Security alert in AAD index and UUID of the  alert in memory (Framework alert) are different.

Ideally they should be the same to avoid confusion.
We can try to pass the uuid to Framework alert (then above code works) or use the below code to hack it and get it work for now.

```
return !summarizedAlerts.all.data.some(
      (alert) =>
        get(alert, ALERT_UUID) === this.getId() || get(alert, ALERT_UUID) === this.getUuid()
    );
```

`get(alert, ALERT_UUID) === this.getId()` is for Security Rule
`get(alert, ALERT_UUID) === this.getUuid()` is for Observability Rule

Reply:
yea looks like we are encountering this: https://github.com/elastic/kibana/issues/144862

i can make this change